### DISCUSSION TITLE
How to get mailmap to work and record contributions in relevant user's profile?

### DISCUSSION THREAD

**Discussion by meetmangukiya** [community/community#22518] [New to GitHub]
<p>Git mailmap is used for patching / correcting emails of a particular contact in git. It replaces and corrects info in git log, etc. How to get github to record contributions to the right person’s profile using the mailmap to get the correct email address to record contributions to?</p>

**Comment by ghost** [comment, marked as answer]
<p>Hi <a class="mention" href="/meetmangukiya">@meetmangukiya</a>. Thanks so much for reaching out on the GitHub community.</p>
<p>Unfortunately at the present time, GitHub does not support using mailmap files. We do have an open feature request for this, and I’ve added your post to this. You are also welcome to submit feedback directly through our <a href="https://support.github.com/contact/feedback">feedback form</a>.</p>
<p>The only way to update the contributor details in your commits is to rewrite git history, which we generally don’t recommend, particularly if your repo has multiple contributors. However, we do have a guide on doing this: <a href="https://docs.github.com/en/github/using-git/changing-author-info">Changing author info</a>.</p>

**Reply by ssbarnea** [reply]
Where is this "open feature request". I tried to click on "feedback form" link but this sends directory to the generic community discussion forum, so not of much use.

That `.mailmap` feature is important for other reasons too: allowing people to remap contributions from defunct email addresses, removing duplicates and as a trick to hide bot contributions under a single "generic" account.

**Reply by adlai** [reply]
[According to Alessandro on stackoverflow](https://stackoverflow.com/a/79353419), it works now.

**Reply by richardebeling** [reply]
@adlai I don't see anything backing up that claim in the SO answer. Alessandro made some commits using a gmail address, e.g. [this one](https://github.com/apache/calcite/commit/8581dba43337e27b2d70f540b91f1bdf66d18dab). They are attributed to him on GitHub, but this is because he verified that email address on his account. You can verify on [this commit](https://github.com/richardebeling/github-mailmap-test/commit/17eee7dbaaa338188993426d7cea240073a1149d), which is also attributed to him without any .mailmap entries. There is no indication that the .mailmap file is involved in what he is observing. My [sample repo](https://github.com/richardebeling/github-mailmap-test) still shows the wrong name that should have been corrected by .mailmap.

**Reply by adlai** [reply]
Thank you for the detailed investigation!

**Comment by gianlucam76** [comment]
<p>Github still appears to not support mailmap. Might I ask if there is an active plan to support this in near future? thank you.</p>

**Reply by beutlich** [reply]
@ghost 👻 Any news on this?

**Reply by jeremyjh** [reply]
Its scheduled for implementation on the 12th of Never

**Reply by adlai** [reply]
> @ghost 👻 Any news on this?

@beutlich the "ghost" highlight will not notify the original responder, as that github account has been deleted. If you feel obnoxiously adventurous, you could try bothering current employees (it's not difficult to figure out who's relevant, from the active discussions), however I recommend against that sort of lobbying behavior, unless you're really polite and already on good terms with the people you wish to engage on this issue.

**Comment by dbaldes** [comment]
It looks like github still doesn't support mailmap. The feature would be greatly appreciated. Thanks!

**Comment by ashvardanian** [comment]
I'd love to see `.mailmap` support as well!

**Comment by alexgwolff** [comment]
any update?

**Comment by Cipscis** [comment]
`.mailmap` isn't just a convenience, there are situations where it's also an important safety feature. I'm surprised this topic has been open for nearly six years and GitHub still doesn't support this.
